lib/content-negotiation.js

```diff
@@ -25,10 +26,12 @@ const SUPPORTED_MEDIA_TYPES = [
     "application/x-www-form-urlencoded",
     "application/cloudevents+json",
 ]; // In preference order
-const textDecoder = new TextDecoder("utf8");
+
+const utf8TextDecoder = new TextDecoder("UTF-8");
 const textEncoder = new TextEncoder();
 
-const jsonUnmarshall = (buffer) => JSON.parse(textDecoder.decode(buffer));
+const jsonUnmarshall = (buffer, charset) =>
+    JSON.parse(resolveDecoder(charset).decode(buffer));
```

Comment:
Should we also check that charset is one of `UTF-x`, per the spec?

Reply:
done